### PR TITLE
doc: note that brew install qt6 is not supported

### PR DESCRIPTION
**Pull request by raulsiles**
Ensure 'qt' v6.x is not installed in brew to force the use to 'qt@5'.

### CODE REVIEW

File: doc/build-osx.md [hunk 1]
```diff
@@ -141,6 +141,15 @@ brew install qt@5
 Note: Building with Qt binaries downloaded from the Qt website is not officially supported.
 See the notes in [#7714](https://github.com/bitcoin/bitcoin/issues/7714).
 
+Note: Be sure 'qt' (version 6.x) is not installed in brew together with 'qt@5'. Only 'qt@5' is required.
```
Comment (jarolrod) [inline doc/build-osx.md:144]: a suggestion

```suggestion
Note: Ensure that the `qt@5` package is installed, not the `qt` package.
```

File: doc/build-osx.md [hunk 2]
```diff
@@ -141,6 +141,15 @@ brew install qt@5
...
+If 'qt' (v6.x) is installed in macOS the build process will fail with a "unknown type name 'QStringRef'; did you mean 'CFStringRef'?" error.
```
Comment (jarolrod) [inline doc/build-osx.md:145]:
```suggestion
If 'qt' is installed, the build process will fail with an error. 
```

File: doc/build-osx.md [hunk 3]
```diff
@@ -141,6 +141,15 @@ brew install qt@5
 Note: Building with Qt binaries downloaded from the Qt website is not officially supported.
 See the notes in [#7714](https://github.com/bitcoin/bitcoin/issues/7714).
 
+Note: Be sure 'qt' (version 6.x) is not installed in brew together with 'qt@5'. Only 'qt@5' is required.
+If 'qt' (v6.x) is installed in macOS the build process will fail with a "unknown type name 'QStringRef'; did you mean 'CFStringRef'?" error. 
+You can use the following two commands to verify if 'qt' (version 6.x) is installed and remove it:
```
Comment (jarolrod) [inline doc/build-osx.md:146]:
```suggestion
if 'qt'  is installed, remove it with the following command:
```

File: doc/build-osx.md [hunk 4]
```diff
@@ -141,6 +141,15 @@ brew install qt@5
...
+
+``` bash
+brew info qt
```
Comment (jarolrod) [inline doc/build-osx.md:149]:
```suggestion

```

**Review by jarolrod — COMMENTED** [review]
I think this note would be ok to add, would prevent any issues from being filed about this in the case someone accidentally installs the `qt` package over `qt@5`.

File: doc/build-osx.md (hunk 4 above)
Comment (raulsiles) [reply doc/build-osx.md:149]: Thanks for the brevity and text updates! :) (Indeed, the goal was to avoid issues about this in advance)

**Comment by raulsiles** [comment]
> I think this note would be ok to add, would prevent any issues from being filed about this in the case someone accidentally installs the `qt` package over `qt@5`.

Not only accidentally. Other brew packages might require 'qt' (v6.x) and, as a result, it could be already installed.

File: doc/build-osx.md (hunk 2 above)
Comment (adamjonas) [reply doc/build-osx.md:145]: This is also tripping up the linter due to the trailing whitespace.

**Comment by fanquake** [comment]
Please [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits). Use a commit message like `doc: note that brew installed qt is not supported`, not the GitHub default message.

**Comment by raulsiles** [comment]
> Please [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits). Use a commit message like `doc: note that brew installed qt is not supported`, not the GitHub default message.

I cannot squash these commits from the GitHub web interface as a contributor (and it will take me some time to switch and setup the repo in the CLI). Please, feel free to squash them directly when merging the pull request as an authorized user. Thanks!

**Comment by hebasto** [comment]
Concept ACK.

Indeed, only `qt@5` package must be installed, because parallel installation of `qt@6` will mess `/usr/local/include/` and `/usr/local/lib/` directories.

**Comment by RandyMcMillan** [comment]
Squashed:

https://github.com/RandyMcMillan/bitcoin-core/commit/c5175a9d19f1c3f3c446b140557425eee76dc70b

**Comment by jarolrod** [comment]
@raulsiles If you don't want to squash this or can't, you can close this and I'll pick this up for you.

**Comment by laanwj** [comment]
Is there a specific reason why our build system cannot handle qt5 and qt6 being both installed? Sure, uninstalling qt6 is a valid workaround but this seems like something that could cause problems if the user is also building things against Qt 6.

**Comment by hebasto** [comment]
> Is there a specific reason why our build system cannot handle qt5 and qt6 being both installed?

I assume the reason is sharing `/usr/local/include/` and `/usr/local/lib/` directories by both qt5 and qt6 installations.

**Pull request closed without merging**